wrappers/hnygrpc/grpc.go

```diff
@@ -72,16 +74,26 @@ func addFields(ctx context.Context, info *grpc.UnaryServerInfo, handler grpc.Una
 	span.AddField("handler.name", handlerName)
 	span.AddField("handler.method", info.FullMethod)
 
-	md, ok := metadata.FromIncomingContext(ctx)
+	pr, ok := peer.FromContext(ctx)
 	if ok {
-		if val, ok := md["content-type"]; ok {
-			span.AddField("request.content_type", val[0])
-		}
-		if val, ok := md[":authority"]; ok {
-			span.AddField("request.header.authority", val[0])
+		if pr.Addr != net.Addr(nil) {
+			span.AddField("request.remote_addr", pr.Addr.String())
 		}
-		if val, ok := md["user-agent"]; ok {
-			span.AddField("request.header.user_agent", val[0])
+	}
+
+	headersToFields := map[string]string{
+		"content-type":      "request.content_type",
+		":authority":        "request.header.authority",
+		"user-agent":        "request.header.user_agent",
+		"x-forwarded-for":   "request.header.x_forwarded_for",
+		"x-forwarded-proto": "request.header.x_forwarded_proto",
+	}
+	md, ok := metadata.FromIncomingContext(ctx)
+	if ok {
+		for headerName, fieldName := range headersToFields {
```

Comment:
I'm rethinking this loop. Nicer (I think) to read, but is a few new memory allocations in instrumentation hot path.

Reply:
I think this loop is good 👍🏻

Reply:
We'll ship the loop! 🔁